Automattic/jetpack
# My Jetpack: Add a Protect 'needs-attention' status when Threats are detected.

This PR adds a 'needs-attention' status to the Protect product card in My Jetpack when Protect threats are detected.
The Primary CTA changes to "Fix threats".
The 'needs-attention' status itself is actually hidden in the sense that the status still remains as "Active", however the product card border will either be yellow or red

With Threats | With Critical Threats
--- | ---
![Screen Shot 2024-12-16 at 09 05 01](https://github.com/user-attachments/assets/4a61f4dd-3630-4c52-83f2-0a97940c38ec) | ![Screen Shot on 2024-12-16 at 09-07-07](https://github.com/user-attachments/assets/8e559882-e43c-4947-94bc-639a1e5e1e78)






Fixes https://github.com/Automattic/jetpack-roadmap/issues/1939

## Proposed changes:

* I split the `needs_attention` status into 2 separate statuses (`needs-attention__warning` & `needs-attention__error`)
* When Protect detects threats, the protect card has yellow border (`needs-attention__warning`), if there are Critical threats, Protect has a red border (`needs-attention__error`).
* If threats are detected, the Protect CTA is "Fix threats".

### Other information:


## Jetpack product discussion



Project Thread: pbNhbs-bJN-p2
Design post: p1HpG7-umC-p2

## Does this pull request change what data or activity we track or use?



## Testing instructions:






- Install/activate Jetpack Protect with a Scan upgrade
- Using the Scan Helper in the Jetpack Debug Helper/Tools plugin, select the first EIACAR Threats selection to Add five EICAR threats to the site (one for every severity).
- Run a scan to detect them
- Proceed to the My Jetpack section, scroll down to the Protect product card.
- Verify the Protect product card has a red border, the CTA is "Fix threats" and the status appears as "Active".
- Now, using the Scan Helper in the Jetpack Debug Helper/Tools plugin again, remove the first EIACAR Threats selection  and add a different threat (like maybe a vulnerable plugin threat).
- Now verify the Protect product card has a yellow border, the CTA is "Fix threats" and the status appears as "Active".
